tools/analyzer_plugin/lib/src/diagnostic/duplicate_prop_cascade.dart

```diff
@@ -31,6 +31,10 @@ class DuplicatePropCascadeDiagnostic extends ComponentUsageDiagnosticContributor
         return '${prop.targetName.name}.${prop.name.name}';
       }
 
+      if (prop.namespace != 'DomPropsMixin') {
+        return '${prop.namespace}.${prop.name.name}';
+      }
+
       return prop.name.name;
     });
```

Comment:
I think a more straightforward solution would be to just always prefix with the type; then we don't have to worry about whether it's in a nested `.dom`.

```dart
final propUsagesByName = groupBy<PropAssignment, String>(usage.cascadedProps, (prop) {
  return '${prop.namespace}.${prop.name.name}';
});

```